### ISSUE TITLE
need resourceIntegrity in vocab and under w3c

### ISSUE THREAD

**Issue by mprorock** [w3c/vc-data-integrity#193]
per pr w3c/vc-data-model#1140 need `resourceIntegrity` in the the right spots in vocab, etc.

@OR13 for more color commentary if needed

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-07-05-vcwg#section2-1) on 2023-07-05

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.1. need resourceIntegrity in vocab and under w3c (issue vc-data-model#1152)

_See github issue [vc-data-model#1152](https://github.com/w3c/vc-data-integrity/issues/193)._




**Manu Sporny:** This is definitely pre-candidate rec.  

**Michael Prorock:** I'm fine raising the PR. I may be delayed two weeks due to travel.  

</details>

**Comment by OR13** [comment]
@iherman perhaps you can assist?

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-08-30-vcwg#section4-1) on 2023-08-30

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 4.1. need resourceIntegrity in vocab and under w3c (issue vc-data-model#1152)

_See github issue [vc-data-model#1152](https://github.com/w3c/vc-data-integrity/issues/193)._




**Brent Zundel:** Raised by Mike Prorock. Ready for PR.  

**Michael Prorock:** I thought we were going to get some guidance from Ivan, but maybe I'm misremembering.  
… it's ready for PR. I'll try to get to it.  

**Sebastian Crane:** is this the same a w3c/vc-data-model#1261.  

**Manu Sporny:** no.  

**Brent Zundel:** no.  

**Michael Prorock:** no. different.  

> *Manu Sporny:* This should be an easy one -- fairly straight-forward PR.

**Brent Zundel:** the resource integrity is literally adding that term to the vocabulary (rather than integrity of the vocabulary).  

</details>

**Comment by iherman** [comment]
Oops. Coming back from my short holidays, I took the various new issues/comments in a fairly random order... I have run into the same issue while answering another issue, see https://github.com/w3c/vc-data-model/issues/1265#issuecomment-1705009825.

TL;DR:  I believe the vocabulary misses two properties: 

> - `digestSRI`'s domain is any Resource (but not a BNode), range is, at first approximation, `xsd:string` (ideally, a datatype that describes exactly what the value looks like).
> - `mediaType`'s domain is any Resource (but not a BNode), range is, at first approximation, `xsd:string` (ideally, a datatype that describes exactly the media type string).
>
>***These two properties are not only missing from the vocabulary, but also missing from the context!***
>
> Note that, for the `mediaType`, an alternative would be to use the term `encodingFormat` instead, referring to http://schema.org/encodingFormat. Do not reinvent the wheel, and all that... That would mean just add an entry to the context file, and this property is oblivious to the vocabulary.

Do I understand the question right, @mprorock @OR13 ?

**Comment by OR13** [comment]
Yep, the need to be added to both, I would do best effort on domain and range, but he key part is that they appear in the context, or they won't benefit the core data models semantics value proposition.

**Comment by msporny** [comment]
Looks like these things exist now:

https://www.w3.org/2018/credentials/#relatedResource
https://www.w3.org/2018/credentials/#digestSRI
https://www.w3.org/2018/credentials/#sriString
https://schema.org/encodingFormat

Closing.
